crates/floresta-chain/src/pruned_utreexo/mod.rs

```diff
@@ -359,11 +363,49 @@ impl<T: BlockchainInterface> BlockchainInterface for Arc<T> {
         proof: Proof,
         inputs: HashMap<OutPoint, TxOut>,
         del_hashes: Vec<sha256::Hash>,
+        chain_time: ChainTime,
     ) -> Result<(), Self::Error> {
-        T::validate_block(self, block, proof, inputs, del_hashes)
+        T::validate_block(self, block, proof, inputs, del_hashes, chain_time)
     }
 
     fn get_fork_point(&self, block: BlockHash) -> Result<BlockHash, Self::Error> {
         T::get_fork_point(self, block)
     }
 }
+
+pub mod chain_time {
+    use core::time::Duration;
+    /// Here we delegate to the consumer of the library to implement the [`ChainTimeInterface`] so we dont need to get stuck with std::time.
+    ///
+    /// Take a look on how we implement the [`ChainTimeInterface`] for the [`ChainTime`] struct if you want to implement your own.
+    pub trait ChainTimeInterface {
+        /// This should return a zero time for test cases. While validating, if we receive a zero time, Time-Related checks should be skipped.
+        fn zero() -> ChainTime;
+        /// This should return a unix timestamp.
+        fn get_time(&self) -> Duration;
+        /// This should return a new instance of ChainTime.
+        fn new(time: Duration) -> ChainTime;
+    }
+    /// We need a some local time notation while validating blocks.
+    ///
+    /// floresta-chain relies on the [`ChainTime`] to have some block validation.
+    #[derive(Debug, Clone, Copy)]
+    pub struct ChainTime(Duration);
+
+    impl ChainTime {
+        /// We use zero for test cases. While validating, if we receive a zero time, Time-Related checks should be skipped.
+        pub const ZERO: ChainTime = ChainTime(Duration::from_secs(0));
+    }
+
+    impl ChainTimeInterface for ChainTime {
+        fn zero() -> ChainTime {
+            ChainTime(Duration::from_secs(0))
+        }
+        fn get_time(&self) -> Duration {
+            self.0
+        }
+        fn new(time: Duration) -> ChainTime {
+            ChainTime(time)
+        }
+    }
+}
```

Comment:
Now we can delegate the time data for the user, Make assumptions or implement it ourselfs

Reply:
I know thats not the most generic approach to implement it, but its the mid-term that i find enough... Trying a generic approach would introduce some bugs